### PR TITLE
Cloud Functions for Users

### PR DESCRIPTION
**Pull request by amanmibra**
## Purpose
Create and set up user-related cloud functions, including auth, setting user data, etc.

## To-Do 
_Please add more if you can think of any_
- [x] Add new user to collection
- [x] Add user-specific info to docs when signing up
- [x] Set up queries to get user data from HTTPS triggers
- [x] Add docs/comments about functions

## How to test
Ask @amanmibra

### CODE REVIEW

File: functions/src/user.ts [hunk 1]
```diff
@@ -0,0 +1,20 @@
+import * as functions from 'firebase-functions';
```
Comment (mafuvadze) [inline functions/src/user.ts:1]: Not a fan of import *
I prefer to import only what you need. This also has slight performance issues

File: functions/src/user.ts [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+import { firebase } from './../config';
+
+const newUserSignIn = functions.auth.user().onCreate(user => {
+  console.log('New user sign in: ', user.data);
+  console.log('New user sign in: ', user.data.providerData);
+  return firebase.firestore().collection('users').add(user.data)
+  .then(ref => {
+    console.log('ref value', ref)
```
Comment (mafuvadze) [inline functions/src/user.ts]: Really bad to have console.logs in production code. I'd think about making a logger. Would also allow us to collect better analytics for errors and warnings in we have a central place to report information/errors/warnings

**Review by mafuvadze — CHANGES_REQUESTED** [review]
Looks good, just a couple nits

File: functions/src/user.ts (hunk 1 above)
Comment (amanmibra) [reply functions/src/user.ts:1]: This is necessary for the time-being as I am new to this library and I'm getting used to what triggers/functionalities of the library is needed and not needed. Once that period of discovery and adjustment has passed, then I will work on importing only the dependencies I need.

Comment (amanmibra) [reply functions/src/user.ts:1]: Thanks for pointing it out

**Comment by amanmibra** [comment]
@mafuvadze this PR is nowhere near finished, I will ping you later when the review is actually needed

**Comment by mafuvadze** [comment]
Sounds good! Just add TODOs so you don't forget to fix them later

**Review by mafuvadze — APPROVED** [review]

**Comment by amanmibra** [comment]
@mafuvadze Pls approve

**Comment by amanmibra** [comment]
Did you actually check the new commits? @mafuvadze

**Comment by amanmibra** [comment]
Please make sure you do a review right before you merge because I made a bunch of changes.

**Pull request merged**